src/dtos/unit.dto.ts

```diff
@@ -93,19 +100,110 @@ export class UnitDTO {
   endDate: Date;
 
   @ApiProperty({
-    description:
-      propertyMetadata.monitorLocationDTONonLoadBasedIndicator.description,
-    example: propertyMetadata.monitorLocationDTONonLoadBasedIndicator.example,
-    name:
-      propertyMetadata.monitorLocationDTONonLoadBasedIndicator.fieldLabels
-        .value,
+    description: propertyMetadata.unitDTOSourceCategoryCd.description,
+    example: propertyMetadata.unitDTOSourceCategoryCd.example,
+    name: propertyMetadata.unitDTOSourceCategoryCd.fieldLabels.value,
   })
   @IsOptional()
-  @IsInt()
-  @IsInRange(0, 1, {
-    message: (args: ValidationArguments) => {
-      return `The value of [${args.value}] for [${args.property}] must be an integer of 0 and 1 for [${KEY}].`;
-    },
+  @IsString()
+  sourceCategoryCd?: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOCommOpDate.description,
+    example: propertyMetadata.unitDTOCommOpDate.example,
+    name: propertyMetadata.unitDTOCommOpDate.fieldLabels.value,
   })
-  nonLoadBasedIndicator: number;
+  @IsOptional()
+  @IsDateString()
+  commOpDate?: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOComrOpDate.description,
+    example: propertyMetadata.unitDTOComrOpDate.example,
+    name: propertyMetadata.unitDTOComrOpDate.fieldLabels.value,
+  })
+  @IsOptional()
+  @IsDateString()
+  comrOpDate?: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOUserId.description,
+    example: propertyMetadata.unitDTOUserId.example,
+    name: propertyMetadata.unitDTOUserId.fieldLabels.value,
+  })
+  @IsNotEmpty()
+  @IsString()
+  userId: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOAddDate.description,
+    example: propertyMetadata.unitDTOAddDate.example,
+    name: propertyMetadata.unitDTOAddDate.fieldLabels.value,
+  })
+  @IsNotEmpty()
+  @IsDateString()
+  addDate: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOUpdateDate.description,
+    example: propertyMetadata.unitDTOUpdateDate.example,
+    name: propertyMetadata.unitDTOUpdateDate.fieldLabels.value,
+  })
+  @IsOptional()
+  @IsDateString()
+  updateDate?: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOOpStatusCd.description,
+    example: propertyMetadata.unitDTOOpStatusCd.example,
+    name: propertyMetadata.unitDTOOpStatusCd.fieldLabels.value,
+  })
+  @IsOptional()
+  @IsString()
+  opStatusCd?: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOStatusBeginDate.description,
+    example: propertyMetadata.unitDTOStatusBeginDate.example,
+    name: propertyMetadata.unitDTOStatusBeginDate.fieldLabels.value,
+  })
+  @IsOptional()
+  @IsDateString()
+  statusBeginDate?: string;
```

Comment:
Why is this date a string?

Reply:
This is explicitly cast to a string in the format `yyyy-mm-dd` in the `getUnitDetails` function of `UnitService` and `UnitWorkspaceService`.